runner_test.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+import unittest
+from app import app
+
+
+class LaunchingTest(unittest.TestCase):
+
+    def setUp(self):
+        self.app = app.test_client()
+        self.app.testing = True
+
+    def test_valid_url(self):
+        result = self.app.get('/')
+        self.assertEqual(result.status_code, 200)
+
+    def test_invalid_url_page_not_found(self):
+        result = self.app.get('/a/a')
+        assert 'Error 404' in str(result.data)
+
+    def test_query_input(self):
+        result = self.app.get('/variable=value')
+        assert u'Your query is variable which is equal to value' in str(result.data)
```

Comment:
problem: what does the `u` in front of the string indicate? what result type does `str` offer?